### DISCUSSION TITLE
Variable not changing when @click.prevent in alpine 3

### DISCUSSION THREAD

**Discussion by slzrdev** [alpinejs/alpine#3822] [7. Bugs (archived)] [answered]
Working in laravel-livewire 2. I have a simple nav menu that opens on click. The variable it's not updated when click on openMenu.
In the chrome inspector it seems that the nav is reacting because it's highlighted, but because the variable doesn't change the menu doesn't opens. I'm not getting any error in the console. I'have tested these in firefox, chrome and safari on mac.

It works with these version:
https://cdn.jsdelivr.net/gh/alpinejs/alpine@v2.7.3/dist/alpine.js
but doesn't with these one:
https://cdn.jsdelivr.net/npm/alpinejs@3.x.x/dist/cdn.min.js

```
<div class="sm:container sm:mx-auto items-center flex flex-col md:flex-row h-20 md:h-auto justify-center md:justify-between" x-data="{ navIsOpen: false }">

    <a href="#" class="pb-1 px-3 md:hidden w-12 h-12 open-menu absolute top-4 left-4 block cursor-pointer" id="openMenu" :key="openMenu" @click.prevent="navIsOpen = true">
        open menu
    </a>

    <a href="/" class=" lg:w-1/4 font-bold text-3xl">Brand Name</a>

    <nav class="pt-20 px-4 pb-4 md:p-0 flex flex-col md:flex-row md:between md:items-center left-0 top-0 justify-between fixed md:static main-nav lg:w-3/4 z-100 bg-white shadow-2xl md:shadow-none md:flex gap-2" :class="navIsOpen ? 'show' : 'hidden'" id="mainNav" :key="mainNav" @click.away="navIsOpen = false">
        <a href="#" class="h-10 w-10 flex items-center justify-center md:hidden  close-menu absolute top-4 left-6" id="closeMenu" @click.prevent="navIsOpen = false">
            close menu
        </a>
        <ul class="md:gap-4 flex flex-col p-0 md:flex-row text-sm">
            <li class="border-b border-gray-100 md:border-none">
                item
            </li>
            <li class="border-b border-gray-100 md:border-none">
                item 2
            </li>

        </ul>
    </nav>
</div>
```

I have another menu in the page, in another scope, not descending of these one that works well with both versions
```
<aside  class="md:order-1  md:w-[300px] md:pr-8 z-50 md:z-10" id="filters" x-data="{ filtersOpen: false }"  @click.away="filtersOpen = false" >
            
    <div class="uppercase p-2 md:hidden  open-filters fixed top-5 right-4 cursor-pointer" id="openFilters" @click.prevent="filtersOpen = true">Filtrar</div>
    
        <div class="col-span-2 fixed top-0 md:top-auto right-0 md:relative overflow-auto md:overflow-visible max-h-screen md:max-h-full shadow-2xl md:shadow-none w-3/4 md:w-full md:block px-4 pt-20 pb-4 md:pr-4 md-pl-0 md:py-0 md:px-0 bg-white"
            :class="filtersOpen ? 'show' : 'hidden'" >
        <span
            class="uppercase p-2 flex items-center justify-center md:hidden  open-filters absolute top-4 right-6"
            id="openFilters" @click.prevent="filtersOpen = false">
            <x-icon.times class="w-6 h-6"/>
        </span>

        <div>FILTERS</div>
    </div>
</aside>
```

I tried to change the variable name, replace the openMenu element anchor with a regular div. I tried to debug the variable by having a div with the variable but it doesn't change.
`<div x-text="navIsOpen"></div>`

I don't know how to search for these bug.

**Comment by ekwoka** [comment]
Use buttons, not a tags for js behaviors.

And you don't need prevent on any of these


Can you make a codepen demoing this?

**Reply by slzrdev** [reply]
Thanks for your answer. I've tried also with button, a regular div, and nothing seems to work. 
I created the following codepen demo, it is my first one so I hope I did it properly.

https://codepen.io/slzrdev/pen/JjxPwgN

**Reply by ekwoka** [reply, marked as answer]
The issue here is just the click.away not the click prevents.

It's being set to True and then immediately back to false.

I don't think it should be, but I don't really use outside that often...

I have some large apps that don't use click outside for anything

**Reply by slzrdev** [reply]
Wow! Thanks a lot for your help, that was the issue exactly. I don't understand why it still worked well when using  v2.x.x, I had this error for more than a year until now. Thanks again!

**Comment by SimoTod** [comment]
You need to use .stop, not .prevent

**Reply by ekwoka** [reply]
While this works, click outside and click away are supposed to not have race conditions with click...

There are tests for it that are passing though...

**Reply by SimoTod** [reply]
I think it only prevents race conditions when you use x-show. If you implement that behavior more manually, like the example above, you have to stop the event or something similar to that.

**Reply by ekwoka** [reply]
That is definitely it.
